zebra/rt_netlink.c

```diff
@@ -655,9 +660,13 @@ int netlink_route_change(struct sockaddr_nl *snl, struct nlmsghdr *h,
 			 ns_id_t ns_id, int startup)
 {
 	int len;
-	vrf_id_t vrf_id = (uint32_t)(ns_id << 16);
+	vrf_id_t vrf_id;
 	struct rtmsg *rtm;
 
+	if (vrf_with_netns)
+		vrf_id = (vrf_id_t) ns_id;
```

Comment:
fix spacing.  I've also noticed several lines that go beyond 80 columns.  Have you run the checkpatch.pl scripts?